src/main/java/com/fasterxml/jackson/databind/deser/std/MapDeserializer.java

```diff
@@ -613,7 +613,19 @@ protected final void _readAndBindStringKeyMap(JsonParser p, DeserializationConte
                 if (useObjectId) {
                     referringAccumulator.put(key, value);
                 } else {
-                    result.put(key, value);
+                    if (result.containsKey(key) && ctxt.isEnabled(StreamReadCapability.DUPLICATE_PROPERTIES)) {
+                        Object oldValue = result.get(key);
```

Comment:
This would also need to be done in the other method, `_readAndBind()` so might make sense to create something similar to what `UntypedObjectDeserializer` has:

```java
    private void _squashDups(final Map<String, Object> result, String key,
            Object oldValue, Object newValue)
    {
        if (oldValue instanceof List<?>) {
            ((List<Object>) oldValue).add(newValue);
            result.put(key, oldValue);
        } else {
            ArrayList<Object> l = new ArrayList<>();
            l.add(oldValue);
            l.add(newValue);
            result.put(key, l);
        }
    }
```

Although... I now realize one possibly significant issue. This functionality should only be applied if the nominal value type is `java.lang.Object` -- otherwise it is unlikely `Map` value type allows both T and `List<T>` which is assumed here.

This check must be made in the primary constructor, using `mapType.getContentType()`, checking something like:

   boolean canSquash = mapType.getContentType().hasRawClass(Object.class);